hopp/tools/design/wind/power_curve_tools.py

```diff
@@ -0,0 +1,174 @@
+    
+    Raises:
+        UserWarning: if `wind_speeds_ms` and `cp_curve` are different lengths.
+
+    Returns:
+        list: turbine power (in kW) at each wind speed in `wind_speeds_ms`
+    """
+
+    if len(wind_speeds_ms) != len(cp_curve):
+        raise UserWarning("The length of the wind speed and coefficient of power vectors must be the same")
+
+    rotor_area = np.pi*((rotor_diameter/2)**2)
+    if isinstance(wind_speeds_ms, list):
+        wind_speeds_ms = np.array(wind_speeds_ms)
+    if isinstance(cp_curve, list):
+        cp_curve = np.array(cp_curve)
+    
+    # power available in the wind (kW)
+    p_wind = 0.5*air_density*rotor_area*(wind_speeds_ms**3)/1e3
+    power_kW = cp_curve*p_wind
+    power_kW = np.where(power_kW > rated_power_kW, rated_power_kW, power_kW)
+    power_kW = np.where(power_kW < 0, 0, power_kW)
+    return list(power_kW)
+
+def estimate_thrust_coefficient(wind_speeds_ms, cp_curve, plot=False, print_output=False):
+    """Calculate thrust coefficient curve (Ct) from power coefficient curve (Cp).
+
+    Args:
+        wind_speeds_ms (List | np.ndarray): list of wind speeds in m/s
+        cp_curve (List | np.ndarray): power curve coefficients (cp) at each wind speed in `wind_speeds_ms`
+        plot (bool, Pptional): whether to plot Cp and Ct curve. Defaults to False.
+        print_output (bool, Optional): Whether to print Cp and Ct curves. Defaults to False.
+    
+    Raises:
+        UserWarning: if `wind_speeds_ms` and `cp_curve` are different lengths.
+
+    Returns:
+        list: thrust curve coefficients (Ct) at each wind speed in `wind_speeds_ms`
+    """
+    
+    # Check that the wind speed and the coefficient of power are the same length
+    if len(wind_speeds_ms) != len(cp_curve):
+        raise UserWarning("The length of the wind speed and coefficient of power vectors must be the same")
+     
+    N_wind = len(wind_speeds_ms)
+    ct_curve = list(np.zeros(N_wind))
+    
+    for i in range(N_wind):
+        # calculate induction factor a
+        # solve C_P = 4 * a * (1-a)**a  -> 4 * a**3 - 8 * a**2 + 4 * a - C_P = 0
+        p = np.zeros(4)
+        p[0] = 4
+        p[1] = -8
+        p[2] = 4
+        p[3] = -cp_curve[i]
+        roots = np.roots(p)
+
+        # Take root that is in range of a -> [0, 0.5]
+        a = roots[np.where(np.logical_and(roots>= 0, roots<= 0.5))]
+
+        # Calculate C_T = 4 * a * (1-a)
+        ct_curve[i] = np.round(4 * a * (1-a), 4)
+        # print(coefficient_of_thrust[i], coefficient_of_power[i]/(1-a))    # Equivalent calculation
+  
+
+    if plot:
+        plot_power_curve(wind_speeds_ms,cp_curve,ct_curve)
+
+    if print_output:
+        print("Coefficient of Thrust (Ct): ")
+        for i in ct_curve:
+            print("-",i)
+        print("Coefficient of Power (Cp): ")
+        for i in cp_curve:
+            print("-",i)
+        print("Wind Speed (v): ")
+        for i in wind_speeds_ms:
+            print("-",i)
```

Comment:
This is going to print in a seemingly unhelpful format, could we do something like the following, or similar? Feel free to push back if I'm missing what the intended use is. Just a note that the `:7.4f` is saying left pad to  make the value 7 characters long (or longer if left side of the decimal is more than 3 digits) with 4 decimal accuracy, so feel free to adjust that as needed, if adopting.

```python
print("Wind  Speed (m/s) | Coefficient of Thrust (Ct) | Coefficient of Power (Cp)")
for ws, ct, cp in zip(wind_speeds, ct_curve, cp_curve):
    print(f"{ws:7.4f} | {ct:7.4f} | {cp:7.4f}"
```